engine/sound/src/sound.cpp

```diff
@@ -1314,26 +1314,26 @@ namespace dmSound
                     break;
                 }
             }
+        }
```

Comment:
The code had ended up within the wrong if-statement, leading to the fact we didn't enter the if-statement again, and the sound voice would stop decoding/mixing.

It's a two step process:
* first, decode as many bytes necessary to fill the requested number of audio frames. Store in `instance->m_FrameCount`
* second, consume those frames by calling `Mix()`, and decrease the `instance->m_FrameCount` again.